### PR TITLE
Added anchors

### PR DESCRIPTION
**Pull request by robhuzzey**
#### What does this PR do? (please provide any background)
Adds anchors to the UI Toolkit

#### What tests does this PR have?
Unit test for the Anchor

#### How can this be tested?
Run `npm test`

#### Screenshots / Screencast
N/A

#### What gif best describes how you feel about this work?
baby see, baby do

#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

**Comment by joestump** [comment]
Took a quick glance and it looks solid to me. @jodiedoubleday and @jackdcrawford can you get some :+1: on this? We shipped 0 last week and I'd like to see that up this week!

**Comment by jackdcrawford** [comment]
On it @joestump

File: src/components/anchor/code/templates/anchorTemplate.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+module.exports = function() {
+
+  if(!this.props.children) {
+    return null;
+  }
+
+  var dataAttributes = this.getDataAttributesFromProps();
+
+  return (
```
Comment (Joezo) [inline src/components/anchor/code/templates/anchorTemplate.jsx]: I would argue anything that is above this line should be in the view, not the template. The pieces of logic above should not change if we decide to switch the template out.

Comment (robhuzzey) [inline src/components/anchor/code/templates/anchorTemplate.jsx]: @Joezo to be honest, I thought we were going to phase out the idea of including templates like this anyway?

**Comment by jackdcrawford** [comment]
:+1: Discussed with @robhuzzey & we will add to the documentation once this is merged.

**Comment by robhuzzey** [comment]
Should there be a revision bump in the package.json? Who merges... is it me? Shout me & let me know.

**Comment by jackdcrawford** [comment]
Releasing is currently manual, I'll release once the docs are updated @robhuzzey. We have a jira to improve the releasing workflow & hopefully automate that that we're pulling into our next sprint. I'll keep you in the loop.

tl;dr; Leave this one with me

**Comment by jodiedoubleday** [comment]
Would be good to have `target` on this

**Comment by robhuzzey** [comment]
@jodiedoubleday do we still use `_target`? I've always seen it as a bad thing because you are dictating the link behaviour for the user (I know I like my links to open in the same window unless I specify it).

**Comment by jodiedoubleday** [comment]
Web developers tend to use, the normal browsing public just want the information to be there.
An example of when someone might need this is when taking someone to read terms and conditions while not taking them out of an original booking flow.

Its Standard HTML so i thought i would add it back in :)

File: src/components/anchor/code/views/anchorView.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return {
+      href: '#'
+    };
+  },
+
+  propTypes: {
+    children: React.PropTypes.node,
+    data: React.PropTypes.object,
+    href: React.PropTypes.string,
+    target: React.PropTypes.oneOf(['self', 'blank', 'parent', 'top']),
```
Comment (robhuzzey) [inline src/components/anchor/code/views/anchorView.jsx]: Should this be required? I wouldn't think we'd always want to pass in a target seeing as it's not a required attribute on the W3 spec for anchor?

File: src/components/anchor/code/templates/anchorTemplate.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+var React = require('react');
+
+module.exports = function() {
+
+  var dataAttributes = this.getDataAttributesFromProps();
+  var target = '_' + this.props.target;
+
+  return (
+    <a className="component-anchor" {...dataAttributes} title={this.props.title} href={this.props.href} onClick={this.props.handleClick} target={target}>
```
Comment (robhuzzey) [inline src/components/anchor/code/templates/anchorTemplate.jsx]: Assuming that target is optional... this could break here as we could end up with `target=_`

**Comment by robhuzzey** [comment]
Just looking at this... there is no test for `target` attribute either

File: src/components/anchor/__tests__/anchor-test.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          TestUtils.Simulate.click(anchorDomNode);
+        });
+
+        it('should call the handler', function(){
+          assert(clickHandler.calledOnce);
+        });
+      });
+    });
+
+    it('should have the correct target from props', function(){
```
Comment (jodiedoubleday) [inline src/components/anchor/__tests__/anchor-test.js]: Test is here
